core/client/app/components/gh-posts-list-item.js

```diff
@@ -22,7 +22,7 @@ export default Component.extend(ActiveLinkWrapper, {
     }),
 
     authorAvatar: computed('post.author.image', function () {
-        return this.get('post.author.image') || this.get('ghostPaths.url').asset('/shared/img/user-image.png');
+        return this.get('post.author.image') || `${this.get('ghostPaths.subdir')}/ghost/img/user-image.png`;
```

Comment:
Yeah, as far as I can tell the images have only ever been used in the admin area, or at least it's been that way since v0.3.x. Theme devs would have had to specifically look for those files in core and then choose to use them instead of providing their own assets.